docs/operate-a-stake-pool/hardening-server.md

```diff
@@ -0,0 +1,310 @@
+```
+
+Reboot your server.
+```shell
+sudo reboot
+```
+
+### 4- Activate Unattended-upgrades for automatic security updates
+
+Next step is to enable unattended upgrades to automatically install security updates (this is the default behavior of unattended-upgrades)
+
+```shell
+sudo apt-get install unattended-upgrades
+sudo dpkg-reconfigure -plow unattended-upgrades
+```
+
+### 5- Generate SSH keys
+
+The next step might be one of the most important things to do, to secure your server. Passwords are not safe to use. Instead, use SSH keys. We are going to create a new SSH key pair on your local machine. I strongly recommend using ED25519 algorithm.
+
+**On your local machine**
+
+- Generate a a new key-pair. You will be asked to enter a filename to save your key
+```shell
+ssh-keygen -t ed25519
+```
+
+- Copy the public key to your server
+```shell
+ssh-copy-id -i $HOME/.ssh/<filename> cardano@server-ip
+```
+
+- Now disconnect from your server, and try to login again from your local machine where your private key is. This time you won't need to enter your password.
+```shell
+ssh cardano@server-ip
+```
+***Important : your private SSH key is what allow you to connect to your server. It is stored on your local machine. You should backup it to a secured cold storage, and never ever share it.***
+
+
+### 6- Hardening SSH configuration
+The next-step is to tweak the sshd_config on your server, to make it much more secured
+
+- Open the /etc/ssh/sshd_config file
+```shell
+sudo nano /etc/ssh/sshd_config
+```
+
+- Then locate and update if needed, all the options below
+
+```shell
+PubkeyAuthentication yes
+PasswordAuthentication no
+PermitRootLogin prohibit-password
+PermitEmptyPasswords no
+X11Forwarding no
+TCPKeepAlive no
+Compression no
+AllowAgentForwarding no
+AllowTcpForwarding no
+KbdInteractiveAuthentication no
+```
+
+- Change the default sshd listening port as well with your own custom port (choose between 1024 - 49150)
+```shell
+Port <custom port number>
+```
+
+- Save and close your /etc/ssh/sshd_config. Validate the configuration
+```shell
+sudo sshd -t
+```
+
+- Then restart sshd
+```shell
+sudo systemctl restart sshd
+```
+
+- Disconnect from your server, and reconnect
+```shell
+ssh cardano@server-ip -p <custom port number>
+```
+
+### 7- Firewall configuration
+We are going to use UFW firewall to secure network access to your Cardano node. We are only focusing on ports that are used for SSH, and Cardano node. If you want to use Grafana and Prometheus for monitoring, please, refer to this [guide](https://developers.cardano.org/docs/operate-a-stake-pool/grafana-dashboard-tutorial), to add appropriate ports to your UFW configuration.
+
+***On your Relay Node :***
+```shell
+sudo ufw default deny incoming
+sudo ufw default allow outgoing
+sudo ufw allow to any proto tcp port <CUSTOM SSHD PORT>
+sudo ufw allow to any proto tcp port <YOUR CARDANO NODE PORT>
+sudo ufw enable
+```
+
+***On your Block Producer Node :***
```

Comment:
this could be Sub-heading or tab

Reply:
Subheading for both "On your Block Producer" and "On your Relay" added